Core/Security/m2crypto/X509Certificate.py

```diff
@@ -0,0 +1,478 @@
+
+    return S_OK(notAfter)
+
+  @executeOnlyIfCertLoaded
+  def getNotBeforeDate(self):
+    """
+      Get not before date of a certificate
+
+      :returns: S_OK( datetime )/S_ERROR
+
+    """
+    return S_OK(self.__certObj.get_not_before().get_datetime())
+
+  # @executeOnlyIfCertLoaded
+  # def setNotBefore(self, notbefore):
+  #   """
+  #     Set not before date of a certificate This method is not meant to be used, but to generate a proxy.
+
+  #     :returns: S_OK/S_ERROR
+  #   """
+  #   self.__certObj.set_not_before(notbefore)
+  #   return S_OK()
+
+  @executeOnlyIfCertLoaded
+  def getSubjectDN(self):
+    """
+      Get subject DN
+
+      :returns: S_OK( string )/S_ERROR
+    """
+    return S_OK(str(self.__certObj.get_subject()))
+
+  @executeOnlyIfCertLoaded
+  def getIssuerDN(self):
+    """
+      Get issuer DN
+
+      :returns: S_OK( string )/S_ERROR
+    """
+    return S_OK(str(self.__certObj.get_issuer()))
+
+  @executeOnlyIfCertLoaded
+  def getSubjectNameObject(self):
+    """
+      Get subject name object
+
+      :returns: S_OK( X509Name )/S_ERROR
+    """
+    return S_OK(self.__certObj.get_subject())
+
+  # The following method is in pyGSI,
+  # but are only used by the pyGSI SSL implementation
+  # So I do not really need them
+
+  # @executeOnlyIfCertLoaded
+  # def getIssuerNameObject(self):
+  #   """
+  #     Get issuer name object
+
+  #     :returns: S_OK( X509Name )/S_ERROR
+  #   """
+  #   return S_OK(self.__certObj.get_issuer())
+
+  @executeOnlyIfCertLoaded
+  def getPublicKey(self):
+    """
+      Get the public key of the certificate
+
+      :returns: S_OK(M2crypto.EVP.PKey)
+
+    """
+    return S_OK(self.__certObj.get_pubkey())
+
+  @executeOnlyIfCertLoaded
+  def getSerialNumber(self):
+    """
+      Get certificate serial number
+
+      :returns: S_OK( serial )/S_ERROR
+    """
+    return S_OK(self.__certObj.get_serial_number())
+
+  @executeOnlyIfCertLoaded
+  def sign(self, key, algo):
+    """
+      Sign the cerificate using provided key and algorithm.
+
+      :param key: M2crypto.EVP.PKey object with private and public key
+      :param algo: algorithm to sign the certificate
+
+      :returns: S_OK/S_ERROR
+    """
+    try:
+      self.__certObj.sign(key, algo)
+    except BaseException as e:
+      return S_ERROR(repr(e))
+
+    return S_OK()
+
+  @executeOnlyIfCertLoaded
+  def getDIRACGroup(self, ignoreDefault=False):
+    """
+      Get the dirac group if present
+
+      If no group is found in the certificate, we query the CS to get the default group
+      for the given user. This can be disabled using the ignoreDefault parameter
+
+      Note that the lookup in the CS only can work for a proxy of first generation,
+      since we search based on the issuer DN
+
+      :param ignoreDefault: if True, do not lookup the CS
+
+      :returns: S_OK(group name/bool)
+    """
+    try:
+      return S_OK(asn1_utils.decodeDIRACGroup(self.__certObj))
```

Comment:
A slightly more general question/comment: Hmm, this isn't particularly safe (as a user can theoretically include any DIRAC group name by writing their own software to generate a proxy). Any code using this function should also then check that the user is really a member of the group: Perhaps that check should be done here to avoid it getting forgotten?

Reply:
It is done already, let me find it for you

Reply:
But indeed, I'll add a comment !

Reply:
Here: https://github.com/DIRACGrid/DIRAC/blob/integration/Core/DISET/AuthManager.py#L107

it actually is a good thing to check the membership at this point because imagine the case where you have two setups (typically, prod and certification). If you are granted a given role in certification, and allowed a proxy for an admin role for example, and then use that proxy to contact the production instance, you want it to be rechecked. 

But I will add a comment, you're right